compat/truffle-compile/index.js

```diff
@@ -198,6 +208,21 @@ var compile = function(sourcePath, sourceText, options, callback, isStale) {
         }
       };
 
+      function findImports(pathName) {
+        try {
+          const absPathName = convertToAbsolutePath(pathName, contractsDirectory, nodeDirectory)
```

Comment:
@smarx I spent some time looking at `truffle-resolve`. Here is my gut feeling: 

`EPMSource`, `NPMSource`, `FSSource` thing is intriguing -  the 2nd is probably with respect to npm source and the last is probably with respect to file system sources that are not npm source. Not sure what the first is about other than it probably has to do with installed contracts. 

But that leads to the second thing:  it isn't very well documented. I think we could use it, but I don't want to risk it right now. When we find we need this EPMSource thing though, we should revisit. 

Feel free to push back and offer your thoughts.

Reply:
I would assume `EPMSource` is about http://www.ethpm.com/.

> I think we could use it, but I don't want to risk it right now.

I'm confused, because I think the risk goes the other way. If we don't understand it, we have no hope of replicating it and should instead just use it. (The spec is "do exactly what Truffle does", right?)

Reply:
This piece of the code we don't want to do what truffle does. The resolver code is in fact still getting used the way it used to in its dependency tracking (via profiler). However it is in the replaced code that we are not using it _additionally_ where it is possible we might. 

One concern I have using this is that as we've seen reporting errors that @tagomaru just looked into and improved would have to be redone again. 

That said, if others want to try this and see how it goes, I am cool with that

I have left comments in the code to suggest how/were to replace. If you ignore the funky names in truffle-resolve "import_path", "imported_from" which I think just refer to the contract source path and the contracts directory (usually the basename of that) and manage to supply the right callback, I think this would be a drop-in replacement short of error handling.

Reply:
@smarx informs:
> I would assume EPMSource is about http://www.ethpm.com/.

Ok - thanks! I'll open an issue to revisit all of this later (unless someone is interested in doing it now.)

Reply:
Issue opened - see https://github.com/ConsenSys/truffle-security/issues/125

Let's move discussion there. And thanks @smarx for (trrying to) keep me (more) honest.